### PR TITLE
Refactor ValidationStep Form

### PR DESCRIPTION
**Pull request by youssef-lr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/9582

### Tests

Add a bank account to your account until you reach the validation step that asks for 3 amounts. 

* Verify that when you submit the form, the inputs are highlighted in red an and error text is shown below them when:

1. The input is empty.
2. The e validation amounts are incorrect, verify also that in this case, the error message coming from the backend is displayed.

* Verify that 'Fix errors' button focuses on the first input with an error.
* Verify that drafts are saved properly: enter valid amounts and refresh the page, make sure the amounts entered still show up.

- [ ] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

With a bank account that is awaiting validation codes, head to https://staging.new.expensify.com/bank-account/BankAccountStep

* Verify that when you submit the form, the inputs are highlighted in red an and error text is shown below them when:

1. The input is empty.
2. The e validation amounts are incorrect, verify also that in this case, the error message coming from the backend is displayed.

* Verify that 'Fix errors' button focuses on the first input with an error.
* Verify that drafts are saved properly: enter valid amounts and refresh the page, make sure the amounts entered still show up.

- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/9680864/191854692-7646fca3-c280-48a7-890e-a167482bc88d.mov

### CODE REVIEW

File: src/ONYXKEYS.js
Comment (luacmartins) [inline src/ONYXKEYS.js:174]: I think we should use a single key for all VBBA form steps like we currently do, instead of a separate key for each step. We should use `REIMBURSEMENT_ACCOUNT` instead, but maybe we can move it to `FORMS.REIMBURSEMENT_ACCOUNT`

File: src/libs/actions/ReimbursementAccount/validateBankAccount.js
Comment (luacmartins) [inline src/libs/actions/ReimbursementAccount/validateBankAccount.js:13]: We are changing this function [here](https://github.com/Expensify/App/pull/10838/files#diff-66ce9adc44f2a3a4b12481610a8bd2827bc7cb71c4d3347f0628521cb9129a1bL415-L420) and this file will no longer exist. The loading/error state changes will be handled in the new API command.

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (luacmartins) [inline src/pages/ReimbursementAccount/ValidationStep.js:60]: We can get both value and key from `_.each`, so there's no need to chain them.

```suggestion
        _.each((value, key) => {
            const filteredValue = this.filterInput(value);
            if (ValidationUtils.isRequiredFulfilled(filteredValue)) {
```

File: src/libs/actions/ReimbursementAccount/validateBankAccount.js
Comment (marcochavezf) [inline src/libs/actions/ReimbursementAccount/validateBankAccount.js:23]: I think we should replace `loading: false` with the new `isLoading: false`, no?

By adding 

```js
Onyx.merge(ONYXKEYS.FORMS.VALIDATION_STEP_FORM, {isLoading: false});
```

right after:

https://github.com/Expensify/App/blob/23ebd43804fc930cd640ef49a7fa874f8928d0bd/src/libs/actions/ReimbursementAccount/validateBankAccount.js#L26-L28

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (luacmartins) [inline src/pages/ReimbursementAccount/ValidationStep.js:63]:
```suggestion
            errors[key] = this.props.translate('common.error.invalidAmount');
```

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (luacmartins) [inline src/pages/ReimbursementAccount/ValidationStep.js:70]: We should still filter these values, no?

```
const amount1 = this.filterInput(this.state.amount1);
const amount2 = this.filterInput(this.state.amount2);
const amount3 = this.filterInput(this.state.amount3);

const validateCode = [amount1, amount2, amount3].join(',');
```

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (luacmartins) [inline src/pages/ReimbursementAccount/ValidationStep.js:146]: AFAIK the button text here is always `'validationStep.buttonText'`, so no need for this logic

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (luacmartins) [inline src/pages/ReimbursementAccount/ValidationStep.js:111]: I don't think this is necessary. See my comment below about the button text.

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (luacmartins) [inline src/pages/ReimbursementAccount/ValidationStep.js:153]: The submit button should be at the bottom of the page:
```suggestion
                        style={[styles.mh5, styles.flexGrow1]}
```

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (marcochavezf) [inline src/pages/ReimbursementAccount/ValidationStep.js:60]: NAB: I think alternatively we can use `_.each` to iterate the pairs of key-value of an object
```suggestion
        _.each(values, (value, inputKey) => {
            if (ValidationUtils.isRequiredFulfilled(this.filterInput(value))) {
```

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (marcochavezf) [inline src/pages/ReimbursementAccount/ValidationStep.js:70]: Are we already getting the filtered values here? (don't we need to use `this.filterInput(values.amountX)` anymore?)

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (luacmartins) [inline src/pages/ReimbursementAccount/ValidationStep.js:167]: No need to pass an empty default value. We should check if amount exists in [filterInput](https://github.com/Expensify/App/blob/23ebd43804fc930cd640ef49a7fa874f8928d0bd/src/pages/ReimbursementAccount/ValidationStep.js#L87), i.e. `let value = amount ? amount.trim() : '';`

**Review by luacmartins — CHANGES_REQUESTED** [review]
Thanks for working on this! Left a few comments.

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (marcochavezf) [inline src/pages/ReimbursementAccount/ValidationStep.js:236]: Q: Where are we using this `prop` in `ValidationStep` component? Shouldn't we add it to `propTypes` and `defaultProps`?

**Review by marcochavezf — CHANGES_REQUESTED** [review]
Looks good! I left a few comments 👍🏽 

Edit: Lol I didn't notice that @luacmartins already left some comments a few mins ago (Also I noticed some of them are similar 😄)

File: src/ONYXKEYS.js
Comment (youssef-lr) [reply src/ONYXKEYS.js:174]: I think this will require change in lots of files, should I create a separate PR for it?

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (youssef-lr) [reply src/pages/ReimbursementAccount/ValidationStep.js:236]: Hmm we're not using it in the file at all, so I'll remove it 👍🏻

File: src/libs/actions/ReimbursementAccount/validateBankAccount.js
Comment (youssef-lr) [reply src/libs/actions/ReimbursementAccount/validateBankAccount.js:13]: Should this PR be put on hold until the change is merged?

Comment (youssef-lr) [reply src/libs/actions/ReimbursementAccount/validateBankAccount.js:13]: Sounds good. I'll push the rest of the requested changes and wait til the PR is merged.

File: src/ONYXKEYS.js
Comment (youssef-lr) [reply src/ONYXKEYS.js:174]: bump @luacmartins, I just noticed that the other external refactors also depend on this key, should I use the `REIMBURSEMENT_ACCOUNT` key that is outside of the `FORMS` key in this PR, until all of the other refactors that depend on it are merged, then we can create a separate PR to move the key under `FORMS`. Let me know what you think!

Comment (luacmartins) [reply src/ONYXKEYS.js:174]: Sounds good to me! Let's go with that.

File: src/libs/actions/ReimbursementAccount/validateBankAccount.js
Comment (luacmartins) [reply src/libs/actions/ReimbursementAccount/validateBankAccount.js:13]: FYI [this PR](https://github.com/Expensify/App/pull/10838) is merged

File: src/libs/actions/BankAccounts.js
Comment (luacmartins) [inline src/libs/actions/BankAccounts.js:204]: Unnecessary space

File: src/libs/actions/BankAccounts.js
Comment (luacmartins) [inline src/libs/actions/BankAccounts.js:176]: Why do we need to add this? Your changes to Form are handling the `errors` key, no?

**Review by luacmartins — COMMENTED** [review]
Left a couple of small comments. Also please merge main to fix the lint error.

**Comment by youssef-lr** [comment]
Thanks for the review @luacmartins! Uploading test screenshots in a bit.

**Review by luacmartins — DISMISSED** [review]

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (marcochavezf) [inline src/pages/ReimbursementAccount/ValidationStep.js:53]: typo :) 
```suggestion
     * @returns {Object}
```

**Review by marcochavezf — CHANGES_REQUESTED** [review]
LGTM, just a minor change :)

**Review by marcochavezf — CHANGES_REQUESTED** [review]
LGTM, just a minor change :)

**Review by marcochavezf — DISMISSED** [review]

**Comment by luacmartins** [comment]
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

File: src/libs/actions/FormActions.js
Comment (luacmartins) [inline src/libs/actions/FormActions.js:26]: Why did we create this? I think that we should update the existing `setErrorMessage` function to conform with the new format we have for errors. We are still early on in the form refactors so it shouldn't be hard to update other instances.

**Review by luacmartins — CHANGES_REQUESTED** [review]

File: src/libs/actions/FormActions.js
Comment (youssef-lr) [reply src/libs/actions/FormActions.js:26]: Got it. I will work on it.

Comment (youssef-lr) [reply src/libs/actions/FormActions.js]: done @luacmartins. Also updated my PR [here](https://github.com/Expensify/App/pull/11076) to use `errors`.

**Review by marcochavezf — DISMISSED** [review]

File: src/libs/actions/FormActions.js
Comment (luacmartins) [inline src/libs/actions/FormActions.js:15]: I think we should rename this to `setErrors` since that better describes this method now and update other usages of it to pass an object instead of string. I think the only other usages are in `Form.stories.js` and seems to use [these args](https://github.com/Expensify/App/blob/12314e75dba6488ef7b14e3e51588fa3dc9bc2a8/src/stories/Form.stories.js#L216)

**Review by luacmartins — CHANGES_REQUESTED** [review]

**Comment by luacmartins** [comment]
Conflicts

**Comment by youssef-lr** [comment]
Yup coming from this [PR](https://github.com/Expensify/App/commit/a224f664118e8c5abade6966cfae1fd5908a3bb2), I'll work on merging the changes into the refactored form.

**Comment by luacmartins** [comment]
@youssef-lr is this ready for another review?

**Comment by youssef-lr** [comment]
@luacmartins yes, I tested it and seems to work well. However, I'm slightly bothered by the flickering of the screen after we submit valid codes. The view switches back to the validation step for a second then switches to the success page. This is happening on main too, see the recording below:

https://user-images.githubusercontent.com/9680864/193114031-96b719fb-8142-4ea2-9f64-712601f794de.mov

**Comment by luacmartins** [comment]
@youssef-lr if this is on main, you could either try to solve the issue in this PR or log another issue and try to solve that separately. Up to you.

**Comment by luacmartins** [comment]
@youssef-lr let me know when the conflicts are resolved and this is ready for another review!

**Comment by youssef-lr** [comment]
> if this is on main, you could either try to solve the issue in this PR or log another issue and try to solve that separately. Up to you.

I will create a separate issue for this. 

> Let me know when the conflicts are resolved and this is ready for another review!

All set!

File: src/libs/actions/FormActions.js
Comment (luacmartins) [inline src/libs/actions/FormActions.js:13]: This should be an object, right?

Comment (youssef-lr) [reply src/libs/actions/FormActions.js]: woops yes, fixed.

**Review by luacmartins — APPROVED** [review]

**Comment by youssef-lr** [comment]
@Justicea83 @marcochavezf bump! :)

**Review by Justicea83 — APPROVED** [review]
LGTM!

**Comment by kavimuru** [comment]
@youssef-lr We are not able to test the last step in the PR `Verify that drafts are saved properly: enter valid amounts and refresh the page, make sure the amounts entered still show up.` . When type 2 in all the fields are also showing error.

https://user-images.githubusercontent.com/43996225/194395135-9e284216-af29-455b-8bcf-7f65abebba40.mp4

**Comment by luacmartins** [comment]
>Verify that drafts are saved properly: enter valid amounts and refresh the page, make sure the amounts entered still show up.

@kavimuru the tester should refresh the page and confirm that the amounts entered are still visible. They do not need to submit the form with these amounts, since the amounts are obviously wrong.

**Pull request merged**
